### ISSUE TITLE
pytest-localserver not migrated from bitbucket?

### ISSUE THREAD

**Issue by oz123** [pytest-dev/pytest#7857]
The documentation still links to it, but the original repository is gone.
Can you please restore the repository?

**Comment by The-Compiler** [comment]
AFAIK it was a mercurial repository which was deleted by Bitbucket, so I don't think there's much we can do (other than upload a local copy to GitHub, if someone has one)

**Comment by oz123** [comment]
I have a relatively an old copy of the repository which can be migrated to git (up to version ~0.3). Would like me to push it here?
Maybe someone has a newer copy_

**Comment by bluetech** [comment]
The repositories have been archived, here is for example the old pytest one: https://bitbucket-archive.softwareheritage.org/projects/py/pytest-dev/pytest.html

**Comment by The-Compiler** [comment]
yep, looks like it's here: https://bitbucket-archive.softwareheritage.org/projects/py/pytest-dev/pytest-localserver.html

There are also some other pytest projects there which didn't seem to make it (`curl-report`, maybe `pep8`, `quickcheck`, maybe more?). Dunno what we should do with those, given that nobody seems to care about them much... Maybe import to GitHub and set as archived?

**Comment by bluetech** [comment]
If they had been under pytest-dev in bitbucket then that sounds like a good idea to me.

They would need to be converted from mercurial to git however.

**Comment by nicoddemus** [comment]
> Maybe import to GitHub and set as archived?

Agreed. 👍 

We need some volunteer(s) to: migrate from Hg to Git, import/add to GitHub (just the repository, we can skip issues/PRs for now) and mark as archived.

**Comment by diazona** [comment]
It would be great to get this going so that pytest-localserver could be updated with a newer certificate, at least. I've been getting errors of the form `ssl.SSLError: [SSL: CA_MD_TOO_WEAK] ca md too weak` with recent Python (3.8.11), related to the certificate included in pytest-localserver.

I can help with a one-time conversion, if that's what it takes, but at the moment I don't think I can make any long-term commitment to help maintain the project, so I don't want to step on the toes of someone who could do that.

**Comment by kloczek** [comment]
So what happens with `pytest-localserver`?
Is it discontinued and whatever was using that module should be updated?
Any replacement?

**Comment by diazona** [comment]
I converted the repository to Git using https://github.com/frej/fast-export and uploaded it to https://github.com/diazona/pytest-localserver. (The conversion _seems_ to have worked although I haven't gone through the history to verify it.)

My hope is that whoever "should" be maintaining this project - possibly the original author @redtoad, or someone else who has the "clout" to make sure their version is or becomes the canonical one  - can resume development on it, using my repo as a starting point if necessary. In a pinch, I could probably do a bit of work on it, e.g. updating the certificate, but like I said I'm probably not in a position right now to make a long-term commitment to maintain the package.

**Comment by kloczek** [comment]
Thank you @diazona :)

**Comment by nicoddemus** [comment]
Thanks for migrating @diazona.

Not sure how to attract maintainers to it, other than the original maintainer. Feel free to edit the README to state that it is looking for now maintainers, and the certificate situation.

**Comment by diazona** [comment]
OK thanks for the suggestion @nicoddemus.

I've been hesitant to add anything new to my version of the repository because I don't want to be involved in a fork that competes against the official one. (Of course there is no official repo for the project right now, but I mean if @redtoad comes back to the project in, say, six months and restores the bitbucket repo, that should unquestionably be the official version, not mine.) I guess an edit to the README file is worth it to clarify the situation, though. I'll do that.

**Comment by coordt** [comment]
I'd like to be a maintainer on the project. I think it is a valuable project and want to help out.

How do I help?

**Comment by nicoddemus** [comment]
I think ideally @diazona and @coordt would agree on being co-maintainers, given that @diazona already ported the repository to GitHub and @coordt is interested in being a maintainer.

How does that sound?

**Comment by redtoad** [comment]
Sorry for being dormant. As you can probably tell, I haven‘t had much time for maintaining this project lately. Thanks BTW to @diazona for porting this to git. That was on my TODO list anyway.

I‘m more than happy in the repo is moved under pytest-dev and if one of you likes to take over as an active maintainer.

**Comment by diazona** [comment]
Thanks for chiming in @redtoad! Given that you're on board, I'm happy to move forward with being a co-maintainer of the project with @coordt. It turns out that I've needed to do a bit of work on the project myself for other things that I'm working on, so I do have an interest in being an active co-maintainer.

Would you be willing to add us as maintainers to the pytest-localserver project on pypi so we can update the page?

Also, if the repo should be moved under pytest-dev, I'm happy to do that. @nicoddemus do you think that makes sense, and if so can you help orchestrate the transfer? (I guess you can fork it under pytest-dev and detach the fork, or if I can get temporary permission to create repositories under pytest-dev I can transfer it, or whatever works - I don't know what makes sense here.)

**Comment by redtoad** [comment]
I've just excavated my PYPI credentials. Just let me know your usernames.

**Comment by diazona** [comment]
Same as here: [diazona](https://pypi.org/user/diazona/)

**Comment by coordt** [comment]
Same as here: [coordt](https://pypi.org/user/coordt/)

**Comment by nicoddemus** [comment]
@diazona the details are here:

https://github.com/pytest-dev/pytest/blob/main/CONTRIBUTING.rst#submitting-plugins-to-pytest-dev

But basically, transfer the repository to me, and I will transfer it to pytest-dev and setup the appropriate permissions. 👍

**Comment by diazona** [comment]
OK sounds good, I'll do that now.

It wasn't clear from the instructions whether I need to also send an email to the mailing list, given that you're already on board with transferring the project, but if I it's necessary I'm happy to do so.

I also have a commit ready(ish) to update the maintainer information but I'm going to want to check it with @coordt and @redtoad, and I don't want to hold up the transfer by waiting for that, so I'm thinking I'll upload it as a pull request after the project gets moved, unless there are objections to that approach.

**Comment by nicoddemus** [comment]
Thanks!

Moved to https://github.com/pytest-dev/pytest-localserver and sent invitations:

![image](https://user-images.githubusercontent.com/1085180/140801944-f6e77749-94b3-4efe-b460-8371b3a3c472.png)

Next steps would be to go over the files in the repository (README for example) and update the link.

I'm closing this for now, further issues are probably better opened in the new repository.

Thanks everyone!

**Comment by diazona** [comment]
Great, thank you!

**Comment by kloczek** [comment]
It would be good to add at least last version tag as well.
